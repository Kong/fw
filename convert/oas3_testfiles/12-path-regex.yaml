# The {something} parameter should be converted to a regex
openapi: 3.0.3
info:
  title: Path parameter test
  version: v1
servers:
  - url: "https://example.com"

paths:
  /demo/{something}/else:
    get:
      tags: [demo]
      operationId: create
      description: This is a demo OAS
      parameters:
        - in: path
          name: something
          required: true
          schema:
            type: integer
            minimum: 1
          description: The user ID
      responses:
        "200":
          description: OK
          content:
            "application/json":
              schema:
                properties:
                  status:
                    type: string
tags:
  - name: demo
