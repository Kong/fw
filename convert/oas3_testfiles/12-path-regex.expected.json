{
  "_format_version": "3.0",
  "services": [
    {
      "host": "example.com",
      "id": "520e4991-2d1a-59b8-bf2e-579cca0969a0",
      "name": "path-parameter-test",
      "path": "",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "e97557be-3897-52e3-8b56-92b703d3fb73",
          "methods": [
            "GET"
          ],
          "name": "path-parameter-test_create",
          "paths": [
            "~/demo/(?<something>[^#?/]+)/else$"
          ],
          "plugins": [],
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_12-path-regex.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_12-path-regex.yaml"
      ]
    }
  ],
  "upstreams": []
}